akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/EventWriter.scala

```diff
@@ -89,34 +94,55 @@ private[akka] object EventWriter {
       extends Command
   final case class WriteAck(persistenceId: Pid, sequenceNumber: SeqNr)
 
-  private case class MaxSeqNrForPid(persistenceId: Pid, sequenceNumber: SeqNr, originalErrorDesc: String)
+  private final case class MaxSeqNrForPid(persistenceId: Pid, sequenceNumber: SeqNr, originalErrorDesc: Option[String])
       extends Command
 
-  private def emptyWaitingForWrite = Vector.empty[(PersistentRepr, ActorRef[StatusReply[WriteAck]])]
   private case class StateForPid(
       waitingForReply: Map[SeqNr, (PersistentRepr, ActorRef[StatusReply[WriteAck]])],
-      waitingForWrite: Vector[(PersistentRepr, ActorRef[StatusReply[WriteAck]])] = emptyWaitingForWrite,
+      waitingForWrite: Vector[(PersistentRepr, ActorRef[StatusReply[WriteAck]])] = Vector.empty,
       writeErrorHandlingInProgress: Boolean = false,
-      currentTransactionId: Int = 0)
+      currentTransactionId: Int = 0,
+      latestSeqNr: SeqNr = 0L,
+      usedTimestamp: Long = 0L,
+      seqNrlookupInProgress: Boolean = false,
```

Comment:
this flag shouldn't be needed, can look at `waitingForSeqNrLookup.nonEmpty` instead